### PR TITLE
Agreement liquidated v2

### PR DESCRIPTION
**Pull request by 0xdavinchee**
- super token agreement liquidated v2 without 3Ps logic yet

### CODE REVIEW

File: packages/ethereum-contracts/contracts/agreements/ConstantFlowAgreementV1.sol [hunk 1]
```diff
@@ -755,25 +757,34 @@ contract ConstantFlowAgreementV1 is
...
             // #1.a.1 yes: then reward = (SD / TD) * RL
             int256 rewardAmount = signedSingleDeposit.mul(totalRewardLeft).div(signedTotalDeposit);
-            token.makeLiquidationPayouts(
+            token.makeLiquidationPayoutsV2(
                 flowParams.flowId,
+                abi.encode("v1", 0), // liquidationTypeData
                 liquidator,
+                bondAccount,
                 flowParams.sender,
                 rewardAmount.toUint256(),
```
Comment (0xdavinchee) [inline packages/ethereum-contracts/contracts/agreements/ConstantFlowAgreementV1.sol:774]: should be signed

File: packages/ethereum-contracts/contracts/superfluid/SuperfluidToken.sol [hunk 2]
```diff
@@ -407,6 +416,76 @@ abstract contract SuperfluidToken is ISuperfluidToken
...
+        if (originalBondAccount == address(0)) {
+            bondAccount = liquidatorAccount;
+        }
+        if (originalBondAccount == address(0) && rewardRecipientAccount == originalBondAccount) {
+            rewardRecipientAccount = liquidatorAccount;
+        }
+
+        _balances[bondAccount] = _balances[bondAccount]
+            .add(penaltyAccountDelta.mul(-1))
+            .add(rewardRecipientAccountDelta.toInt256().mul(-1));
```
Comment (0xdavinchee) [inline packages/ethereum-contracts/contracts/superfluid/SuperfluidToken.sol:446]: don't flip sign

File: packages/ethereum-contracts/contracts/superfluid/SuperfluidToken.sol [hunk 3]
```diff
@@ -407,6 +416,76 @@ abstract contract SuperfluidToken is ISuperfluidToken
...
+
+        _balances[penaltyAccount] = _balances[penaltyAccount]
+            .add(penaltyAccountDelta);
+
+        // NOTE: We need to set the rewardRecipientAccount to bondAccount
+        // if the bondAccount is not the 0 address and it is not a bailout
+        // we must do this prior to the 3Ps because the test cases expect the
+        // bondAccount to always receive the reward
+        if (originalBondAccount != address(0) && penaltyAccountDelta <= 0) {
```
Comment (0xdavinchee) [inline packages/ethereum-contracts/contracts/superfluid/SuperfluidToken.sol:455]: bond => reward

File: packages/ethereum-contracts/contracts/superfluid/SuperfluidToken.sol [hunk 4]
```diff
@@ -407,6 +416,76 @@ abstract contract SuperfluidToken is ISuperfluidToken
             );
         }
     }
+    
+    /// @dev ISuperfluidToken.makeLiquidationPayoutsV2 implementation
+    function makeLiquidationPayoutsV2(
+        bytes32 id,
+        bytes memory liquidationTypeData,
+        address liquidatorAccount,
```
Comment (0xdavinchee) [inline packages/ethereum-contracts/contracts/superfluid/SuperfluidToken.sol:424]: add comments for explaining the different accounts + rewardRecipientAccount => rewardAccount

File: packages/ethereum-contracts/contracts/superfluid/SuperfluidToken.sol [hunk 5]
```diff
@@ -407,6 +416,76 @@ abstract contract SuperfluidToken is ISuperfluidToken
...
+        address rewardRecipientAccount,
+        address penaltyAccount,
+        uint256 rewardRecipientAccountDelta,
```
Comment (0xdavinchee) [inline packages/ethereum-contracts/contracts/superfluid/SuperfluidToken.sol:427]: rewardRecipientAccountDelta => rewardAccountBalanceDelta

**Comment by hellwolf** [comment]
abandoned in favor of #574

**Pull request closed without merging**